### DISCUSSION TITLE
Installing Firedrake on a cluster

### DISCUSSION THREAD

**Discussion by MatiasAlday** [firedrakeproject/firedrake#2421] [Firedrake support] [answered]
Hello! I'm currently using Thetis which works with the Firedrake environment. I need to install now Firedrake on a cluster, where I don't have any sudo rights, so I guess I have to create an install script where I need to load modules to match the required dependencies of Firerake.  Could you give me a hand with this please? What are the dependencies required? 

As far as I understand firedrake is based on petsc, and the firedrake-install program asks petsc to install MPI. If I can load a petsc module on the cluster, is there a way to flag firedrake-install to provide my own petsc? (dismiss this question if it is not the way to go)

Another question coming from one of the guys in charge of the cluster:  is it possible to  configure firedrake to use a given MPI, in particular the one provided by the openmpi and py-mpi4py modules ?

Thanks for your help!

 Matias Alday

**Comment by JDBetteridge** [comment, marked as answer]
We are currently developing an spack install to make installing on a cluster easier. If you are happy to give it a try, the current documentation for installing via this method is available [here](https://hackmd.io/3XmDSTm2RHeL_FnqCQ0l_A), and will be moved to the Firedrake wiki once we have ironed out any bugs.

To address some of the specifics above:
- The spack installer does not require `sudo`.
- It's difficult to know what modules you will need, but hopefully your cluster has some documentation on this. You will need to load the latest GCC compiler available and the system MPI distribution at the very least. You can usually get an overview of available modules with `module avail`.
- You shouldn't need to worry too much about dependencies, spack should take care of installing them for you. If you want to use system packages you must ensure they are loaded and added to `~/.spack/packages.yaml`.
- It is possible to use the system OpenMPI if you add it as an external package to the spack configuration and use `^openmpi` in place of `^mpich`

All the steps are detailed in those instructions, but they are a work in progress, so please feel free to ask questions and report back if anything isn't working.

**Reply by MatiasAlday** [reply]
HI! thanks for the quick reply. I'll check the documentation ASAP
